tests/unit/payment-methods/test-class-upe-payment-gateway.php

```diff
@@ -1444,6 +1455,58 @@ public function test_correct_payment_method_title_for_order() {
 		}
 	}
 
+	public function test_correct_payment_method_title_for_order_when_set_for_email() {
+		$payment_methods = [
+			'cheque' => 'Check payments',
+			'cod'    => 'Cash on delivery',
+			'bacs'   => 'Direct bank transfer',
+		];
+
+		// Emulates order creation which sets the payment method and title.
+		$order = WC_Helper_Order::create_order();
+
+		foreach ( $payment_methods as $method => $title ) {
+			$order->set_payment_method( $method );
+			$order->set_payment_method_title( $title );
+
+			$this->mock_upe_gateway->set_payment_method_title_for_email( $order );
+			$this->assertEquals( $title, $order->get_payment_method_title() );
+		}
+	}
+
+	public function test_set_payment_method_title_for_email_fallback() {
+		$this->mock_order_service
+			->expects( $this->once() )
+			->method( 'get_payment_method_id_for_order' )
+			->will(
+				$this->returnValue( '' )
+			);
+
+			$order = WC_Helper_Order::create_order();
+			$order->set_payment_method( 'woocommerce_payments' );
+			$order->set_payment_method_title( 'Popular Payment Methods' );
+
+			$this->mock_upe_gateway->set_payment_method_title_for_email( $order );
+			$this->assertEquals( 'WooPayments', $order->get_payment_method_title() );
+	}
+
+	public function test_set_payment_method_title_for_email_only_runs_for_legacy_upe() {
```

Comment:
I've left https://github.com/Automattic/woocommerce-payments/pull/7285#discussion_r1342647077 wondering if only legacy UPE should run this logic. Depending on what will be the outcome there, this test may be removed.

Reply:
Keeping the logic as is (see https://github.com/Automattic/woocommerce-payments/pull/7285#discussion_r1343711653).